examples/WordCount/Program.cs

```diff
@@ -0,0 +1,707 @@
+// Copyright 2020 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using Confluent.Kafka;
+using Confluent.Kafka.Admin;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using RocksDbSharp;
+
+
+namespace Confluent.Kafka.Examples.WordCount
+{
+    public class ProducerState<K,V>
+    {
+        public IProducer<K,V> Producer { get; set; }
+        public Offset Offset { get; set; }
+    }
+
+    public class Program
+    {
+        /// <summary>
+        ///     Default timeout used for transaction related operations.
+        /// </summary>
+        static TimeSpan DefaultTimeout = TimeSpan.FromSeconds(30);
+
+        /// <summary>
+        ///     Timeout to use after a cancellation request when attempting to
+        ///     cleanly finalize a transaction.
+        /// </summary>
+        static TimeSpan DuringCancellationTimeout = TimeSpan.FromSeconds(2);
+
+        /// <summary>
+        ///     The numer of partitions we'll use for each topic.
+        /// </summary>
+        const int NumPartitions = 12;
+
+        /// <summary>
+        ///     The name of the Kafka topic containing the input lines of text to count the words in.
+        ///         * key = null
+        ///         * value = a line of text (type: string)
+        /// </summary>
+        const string Topic_InputLines = "line-input";
+
+        /// <summary>
+        ///     The name of the Kafka topic containing the stream of words to count.
+        ///         * key = individual words (type: string)
+        ///         * value = null
```

Comment:
don't agree with that because it means storing unnecessary data, which seems much worse than protecting against what i perceive as a second tier possible point of confusion.

Reply:
Problem is that would the topic be compacted, which is not unreasonable, the semantics of this message is the reverse: it performs a delete.
We know this, but our users might not, and they copy this code and go about their business until one day the code is deployed for a compacted topic and suddenly they have data loss.

Re storing unnecessary data: the message framing is around 20-30 bytes per message, a single byte "1" string value doesnt make any difference.